### ISSUE TITLE
NoSuchMethodError: The getter 'inputs' was called on null

### ISSUE THREAD

**Issue by Monotoba** [dart-lang/build#2176]
* Dart VM version: 2.2.0 (Unknown timestamp) on "linux_x64"

* built_collection: '>=2.0.0 <5.0.0'
*  built_value: '>=5.5.5 <7.0.0'
* build_runner: ^1.0.0
* built_value_generator: ^6.4.0
* Article 
* OS = Linux

The code is from "The Boring Show" episodes 1 - 3 found on Google's Youtube Channel.

---------------------------------------------------------------------------------------
The Code:
---------------------------------------------------------------------------------------
import 'dart:convert' as json;
//import 'package:built_value/built_value.dart';
import 'package:built_collection/built_collection.dart';

part 'json_parsing.g.dart';

abstract class Article implements Built<Article, ArticleBuilder> {
  int get id;
  bool get deleted;
  String get type;
  String get by;
  int get time;
  String get text;
  bool get dead;
  int get parent;
  int get poll;
  BuiltList<int> get kids;
  String get url;
  int get score;
  String get title;
  BuiltList<int> get parts;
  int get descendants;

  Article._();

  factory Article([void Function(ArticleBuilder) updates]) = _$Article;
}

List<int> parseTopStories(String jsonStr) {
  return [];
//  final parsed = json.jsonDecode(jsonStr);
//  final listOfIds = List<int>.from(parsed);
//  return listOfIds;
}

Article parseArticle(String jsonStr) {
  return null;
//  final parsed = json.jsonDecode(jsonStr);
//  Article article = Article.fromJson(parsed);
//  return article;
}

---------------------------------------------------------------------------------------------
The Error Messager:
---------------------------------------------------------------------------------------------
~/projects/flutter/hn_app$ flutter packages pub run build_runner build
[INFO] Generating build script...
[INFO] Generating build script completed, took 334ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...

You have hit a bug in build_runner
Please file an issue with reproduction steps at https://github.com/dart-lang/build/issues

NoSuchMethodError: The getter 'inputs' was called on null.
Receiver: null
Tried calling: inputs
dart:core                                                                                                                 Object.noSuchMethod
../../../flutter/.pub-cache/hosted/pub.dartlang.org/build_runner_core-3.0.2/lib/src/asset_graph/serialization.dart 67:20  _AssetGraphDeserializer.deserialize
package:build_runner_core/src/asset_graph/graph.dart 42:48                                                                new AssetGraph.deserialize
package:build_runner_core/src/generate/build_definition.dart 215:38                                                       _Loader._tryReadCachedAssetGraph.<fn>
package:build_runner_core/src/logging/logging.dart 25:30                                                                  logTimedAsync
package:build_runner_core/src/generate/build_definition.dart 213:12                                                       _Loader._tryReadCachedAssetGraph
package:build_runner_core/src/generate/build_definition.dart 79:28                                                        _Loader.prepareWorkspace
package:build_runner_core/src/generate/build_definition.dart 64:50                                                        BuildDefinition.prepareWorkspace
package:build_runner_core/src/generate/build_impl.dart 117:49                                                             BuildImpl.create
package:build_runner_core/src/generate/build_runner.dart 33:42                                                            BuildRunner.create
package:build_runner/src/generate/build.dart 98:35                                                                        build
package:build_runner/src/entrypoint/build.dart 28:24                                                                      BuildCommand.run
package:args/command_runner.dart 196:27                                                                                   CommandRunner.runCommand
package:args/command_runner.dart 111:29                                                                                   CommandRunner.run.<fn>
dart:async                                                                                                                new Future.sync
package:args/command_runner.dart 111:11                                                                                   CommandRunner.run
package:build_runner/src/entrypoint/run.dart 23:31                                                                        run
.dart_tool/build/entrypoint/build.dart 21:22                                                                              main

pub finished with exit code 1

**Comment by natebosch** [comment]
If you could look for the file `.dart_tool/build/<some hash>/asset_graph.json` and attach it to the bug #1804 that would help us a lot - so far we have been unable to reproduce this or gather enough information to debug fully.

The way to get unblocked is to run `flutter packages pub run build_runner clean` - but only do this after attaching the asset graph file if you are able since that command will delete it.

I'll close for now as a duplicate of #1804
